ecs_pod.go

```diff
@@ -15,7 +15,10 @@ type ECSPod interface {
 	Resources() ECSPodResources
 	// StatusInfo returns the current cached status information for the pod.
 	StatusInfo() ECSPodStatusInfo
-	// TODO (EVG-15161): add GetLatestStatus method to get non-cached status.
+	// LatestStatusInfo returns the latest non-cached status information for the
+	// pod. Implementations should query ECS directly for its most up-to-date
+	// status.
+	LatestStatusInfo(ctx context.Context) (*ECSPodStatusInfo, error)
```

Comment:
Why do we return a pointer to `ESCPodStatusInfo`?

Reply:
My standard way of doing things (for better or for worse) is to follow the Go convention that returning a pointer + error means the result value is optional (not great since pointer != optional, but Go lacks real optionals). Returning the pointer makes it easier to understand from the interface that either the method returns the result (and nil error) or an error (and nil result), but not both.